src/updater.js

```diff
@@ -0,0 +1,65 @@
+var Frame = require('../src/frame.js');
+var TenthFrame = require('../src/tenthFrame.js');
+
+function Updater() {
+}
+
+Updater.prototype.update = function(frames) {
+    var that = this;
+    frames.forEach(function(frame, index){
+        if (index !== 9) {
+            that.updateFrame(that, frame, index, frames);
+        }
+    });
+}
+
+Updater.prototype.updateFrame = function(that, frame, index, frames){
+    var next = frames[index + 1];
+    var nextNext = frames[index + 2];
+    if (frame.isStrike()) {
+       that.updateStrike(frame, next, nextNext, index);
+    } else if (frame.isSpare()) {
+       that.updateSpare(frame, next);
+    }
+}
+
+Updater.prototype.updateStrike = function(frame, next, nextNext, index) {
```

Comment:
it's really hard for me to guess what this function does from the name and argument names alone.